test/Microsoft.Framework.PackageManager.FunctionalTests/DnuPackTests.cs

```diff
@@ -0,0 +1,157 @@
+            }
+        }
+
+        [Theory]
+        [MemberData("RuntimeComponents")]
+        public void DnuPack_FrameworkSpecified(string flavor, string os, string architecture)
+        {
+            string expectedDNX =
+            @"Building {0} for DNX,Version=v4.5.1
+  Using Project dependency {0} 1.0.0
+    Source: {1}{2}project.json";
+            string expectedNupkg =
+            @"{0} -> {1}{2}bin{2}Debug{2}{0}.1.0.0.nupkg
+{0} -> {1}{2}bin{2}Debug{2}{0}.1.0.0.symbols.nupkg
+
+Build succeeded.
+    0 Warnings(s)
+    0 Error(s)
+";
+            string stdOut;
+            string stdError;
+            var runtimeHomeDir = TestUtils.GetRuntimeHomeDir(flavor, os, architecture);
+            int exitCode;
+
+            using (var testEnv = new DnuTestEnvironment(runtimeHomeDir))
+            {
+                File.WriteAllText($"{testEnv.RootDir}/project.json",
+                @"{
+                    ""frameworks"": {
+                        ""dnx451"": {
+                        },
+                        ""dnxcore50"": {
+                            ""dependencies"": {
+                                ""System.Runtime"":""4.0.20-*""
+                            }
+                        }
+                    }
+                  }");
+                DnuTestUtils.ExecDnu(runtimeHomeDir, "restore", "", out stdOut, out stdError, environment: null, workingDir: testEnv.RootDir);
+                exitCode = DnuTestUtils.ExecDnu(runtimeHomeDir, "pack", "--framework dnx451", out stdOut, out stdError, environment: null, workingDir: testEnv.RootDir);
+
+                Assert.Empty(stdError);
+                Assert.Contains(string.Format(expectedDNX, Path.GetFileName(testEnv.RootDir), testEnv.RootDir, Path.DirectorySeparatorChar), stdOut);
+                Assert.DoesNotContain("DNXCore", stdOut);
+                Assert.Contains(string.Format(expectedNupkg, Path.GetFileName(testEnv.RootDir), testEnv.RootDir, Path.DirectorySeparatorChar), stdOut);
+                Assert.Equal(0, exitCode);
+                Assert.True(Directory.Exists($"{testEnv.RootDir}/bin"));
+            }
+        }
+
+        [Theory]
+        [MemberData("RuntimeComponents")]
+        public void DnuPack_OutPathSpecified(string flavor, string os, string architecture)
+        {
+            string expectedNupkg =
+            @"{0} -> {1}{2}CustomOutput{2}Debug{2}{0}.1.0.0.nupkg
+{0} -> {1}{2}CustomOutput{2}Debug{2}{0}.1.0.0.symbols.nupkg
+
+Build succeeded.
+    0 Warnings(s)
+    0 Error(s)
+";
+            string stdOut;
+            string stdError;
+            var runtimeHomeDir = TestUtils.GetRuntimeHomeDir(flavor, os, architecture);
+            int exitCode;
+
+            using (var testEnv = new DnuTestEnvironment(runtimeHomeDir))
+            {
+                File.WriteAllText($"{testEnv.RootDir}/project.json",
+                @"{
+                    ""frameworks"": {
```

Comment:
Use the `Project` object and generate the json from it instead of hardcoding json here